fhirAssembly/src/main/resources/assembly/com.vantiq.fhir.fhirConnection/procedures/com/vantiq/fhir/fhirService_checkAuthenticationSetup.vail

```diff
@@ -13,13 +13,32 @@ if (theConfig == null) {
 
 var oauthConfig = SELECT ONE FROM sources WITH configure = true WHERE name == "com.vantiq.fhir.oauthSource"
 if (oauthConfig == null || oauthConfig.config == null) {
-    exception("com.vantiq.fhir.source.noconfig", "No configuration was found for source {0}.", ["com.vantiq.fhir.oauthSource"])
+    exception("com.vantiq.fhir.source.noconfig", "No configuration was found for source {0}.",
+        ["com.vantiq.fhir.oauthSource"])
 }
 
 if (authMechanism == "Basic") {
-    if (theConfig.config.username == null || theConfig.config.password == null) {
+    if (theConfig.config.accessToken == null &&
+            (theConfig.config.username == null || theConfig.config.password == null)) {
         exception("com.vantiq.fhir.basicauth.required",
-            "A username and password are required for Basic authentication: {0}", [theConfig])
```

Comment:
Tweak auth info validation with new items.

Added AHA for the ability to check some of these more dynamic things via rules or conditionals for configuration parameters.  By the time we check here it's really too late, but it's the only chance we get.